metldata/model_utils/assumptions.py

```diff
@@ -15,14 +15,64 @@
 #
 
 """Logic to check basic assumptions about the metadata model."""
-from metldata.model_utils.anchors import InvalidAnchorPointError, get_anchor_points
+from typing import Optional
+
+from metldata.model_utils.anchors import (
+    AnchorPointNotFoundError,
+    InvalidAnchorPointError,
+    filter_anchor_points,
+    get_anchors_points_by_target,
+)
 from metldata.model_utils.essentials import ROOT_CLASS, MetadataModel
 
 
 class MetadataModelAssumptionError(RuntimeError):
     """Raised when the assumptions about the metadata model are not met."""
 
 
+def check_class_exists(model: MetadataModel, class_name: str) -> None:
+    """Check that a class with the given name exists.
+
+    Raises:
+        MetadataModelAssumptionError: if validation fails.
+    """
+
+    # has a class called class_name:
+    class_ = model.schema_view.get_class(class_name=class_name)
+
+    if class_ is None:
+        raise MetadataModelAssumptionError(
+            f"A class called {class_name} is required but does not exist."
+        )
+
+
+def check_class_slot_exists(
+    model: MetadataModel, class_name: str, slot_name: str, ignore_parents: bool = False
+) -> None:
+    """Check that a class with the given name exists and that it has a slot with the
+    given name. If ignore parents is set to True, slots that are inherited from parent
+    or mixinins are ignored.
+
+    Raises:
+        MetadataModelAssumptionError: if validation fails.
+    """
+
+    check_class_exists(model=model, class_name=class_name)
+
+    all_slots = model.schema_view.class_slots(
+        class_name=class_name, direct=ignore_parents
+    )
+
+    if slot_name not in all_slots:
+        raise MetadataModelAssumptionError(
+            f"A slot called {slot_name} is required but does not exist"
+            + f" in the {class_name} class."
```

Comment:
```suggestion
            f"A slot called '{slot_name}' is required but does not exist"
            + f" in the '{class_name}' class."
```

Since you're using the single quotes in other error messages as well.

I sometimes also do this instead: f"A slot called {slot_name!r} is required..." which will show `repr(slot_name)`, i.e. automatically add quotes and show non-strings more nicely.